src/consensus/consensus.rs

```diff
@@ -336,6 +413,55 @@ impl BlockProposer {
         }
         confirmed_shard_chunks
     }
+
+    async fn publish_new_block(&self, block: Block) {
+        for tx in &self.block_tx {
+            let _ = tx.send(block.clone()).await;
+        }
```

Comment:
A better option might be to use a broadcast channel for this https://docs.rs/tokio/latest/tokio/sync/broadcast/index.html

Reply:
I ended up using a lock as a temporary solution til we have the db to avoid the messiness around channels. Once we have the db in, we won't need a broadcast channel.